webapp/blueprints/users.py

```diff
@@ -123,6 +127,46 @@ def login_post():
 @users.route('/login', methods=['GET'])
 def login_get():
     return flask.render_template('login.html')
+ 
+@users.route('/forgot_password', methods=['GET'])
+def forgot_password_get():
+  return flask.render_template('forgot_password.html')
+
+
+@users.route('/forgot_password', methods=['POST'])
+def forgot_password_post():
+  
+  email = flask.request.form.get('email')
+
+  if not email :
+        flask.flash("You need to provide an email adress")
+        return flask.redirect(flask.url_for('users.forgot_password_get'))
+  
+  user = User.query.get(email)
+  
+  if user is None :
+      flask.flash("Mail not found")
+      return flask.redirect(flask.url_for('users.forgot_password_get'))
+      
+  letters = string.ascii_lowercase + string.digits
+  password = ''.join(random.choice(letters) for i in range(5))
```

Comment:
not very strong! use upercase letters and at least 8 characters (or even 32, whatever the length of the password, the user will copy-paste it, so you don't need to have something easy to remember).

Reply:
Moreover, if you look carefully at the definition password handler, you already have a mechanism to generate a random token and make the user define a new password.